linkerd/app/outbound/src/tcp/logical.rs

```diff
@@ -1,12 +1,45 @@
-use super::{Concrete, Logical};
-use crate::Outbound;
-use linkerd_app_core::{io, profiles, proxy::api_resolve::ConcreteAddr, svc, Error};
-use tracing::debug_span;
+use super::{Concrete, Logical, Outbound};
+use linkerd_app_core::{
+    io,
+    profiles::{self, Profile},
+    proxy::api_resolve::ConcreteAddr,
+    svc, Error,
+};
+use linkerd_distribute as distribute;
 
-#[cfg(test)]
-mod tests;
+#[derive(Debug, thiserror::Error)]
+#[error("no route")]
+pub struct NoRoute;
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct Params {
+    logical: Logical,
+    route: RouteParams,
+    backends: distribute::Backends<Concrete>,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct RouteParams {
+    logical: Logical,
+    distribution: Distribution,
+}
+
+type BackendCache<N, S> = distribute::BackendCache<Concrete, N, S>;
+type Distribution = distribute::Distribution<Concrete>;
+
+// === impl Outbound ===
 
 impl<N> Outbound<N> {
+    /// Builds a `NewService` that produces a router service for each logical
+    /// target.
+    ///
+    /// The router uses discovery information (provided on the target) to
+    /// support per-request connection routing over a set of concrete inner
+    /// services. Only available inner services are used for routing. When
+    /// there are no available backends, requests are failed with a
+    /// [`svc::stack::LoadShedError`].
```

Comment:
nit, take it or leave it: if we're documenting the errors this returns, should we mention that `NoRoute` errors are returned when no route matches?